### PR TITLE
Dev nxtflasher 531

### PR DESCRIPTION
**Pull request by FrikadellenFred**
pull request used for review

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -425,6 +457,37 @@ if 'NETX90' in atPickNetxForBuild:
     env_netx90_dbg = env_netx90_default.Clone()
     env_netx90_dbg.Append(CPPDEFINES = [['CFG_DEBUGMSG', '1']])
     elf_netx90_dbg, bin_netx90_dbg, lib_netx90_dbg = flasher_build('flasher_netx90_debug', env_netx90_dbg, 'targets/netx90_dbg', src_lib_netx90, src_main_netx90)
+    
+    cwd = os.getcwd()
+    strKeyRomPath = "keys/keys.xml" # must be copied to repo
```
Comment (FrikadellenFred) [inline SConstruct:462]: does it even work without keys.xml?

File: templates/flasher.lua [hunk 2]
```diff
@@ -186,8 +214,50 @@ function get_flasher_binary_path(iChiptype, strPathPrefix, fDebug)
...
+    if iChiptype==romloader.ROMLOADER_CHIPTYP_NETX90 or iChiptype==romloader.ROMLOADER_CHIPTYP_NETX90B or
+            iChiptype==romloader.ROMLOADER_CHIPTYP_NETX90C or iChiptype==romloader.ROMLOADER_CHIPTYP_NETX90D then
+		strNetxName = 'netx90'
+        strSubdirName = 'netx90' -- will be an other name in the future
+    else
+        strNetxName = nil
+    end
+
+	if not strNetxName then
+		error("Unknown or unsupported chiptyp! " .. tostring(iChiptype))
```
Comment (FrikadellenFred) [inline templates/flasher.lua:249]: unsupported chip type for hboot mode

File: templates/flasher.lua [hunk 3]
```diff
@@ -186,8 +214,50 @@ function get_flasher_binary_path(iChiptype, strPathPrefix, fDebug)
...
+    end
+    strPathDir = path.join(strSecureOption, strSubdirName)
```
Comment (FrikadellenFred) [inline templates/flasher.lua:251]: check if strNetxName  is not nil before trying to create the path

File: templates/flasher.lua [hunk 4]
```diff
@@ -200,14 +270,22 @@ end
...
+    if bHbootFlash == true then
+        -- offset when parsing the hboot image instead of the binary
+        ulExtraOffset = 0x400
+        aAttr.ulLoadAddress = 0x200C0  -- load address for images in hboot mode
+    else
+        aAttr.ulLoadAddress = get_dword(strData,32 + 1)
+    end
+
 	-- Get the load and exec address from the binary.
-	aAttr.ulLoadAddress = get_dword(strData, ${OFFSETOF_FLASHER_VERSION_STRUCT_pulLoadAddress} + 1)
```
Comment (FrikadellenFred) [inline templates/flasher.lua:205]: use placeholder since it is still the template

File: templates/flasher.lua [hunk 5]
```diff
@@ -256,10 +334,23 @@ end
...
-	local strPath = get_flasher_binary_path(iChiptype, strPrefix, fDebug)
+    local strPath
+
+    local usMiVersionMin = get_mi_version_min(tPlugin)
+    local usMiVersionMaj = get_mi_version_maj(tPlugin)
+
+    print(string.format("usMiVersionMaj 0x%04x", usMiVersionMaj))
+    print(string.format("usMiVersionMin 0x%04x", usMiVersionMin))
+
+    if (usMiVersionMaj == 3 and usMiVersionMin >=1 or usMiVersionMaj > 3) and bCompMode == false then
```
Comment (FrikadellenFred) [inline templates/flasher.lua:348]: noch eine klammer hinzufügen

File: SConstruct (hunk 1 above)
Comment (dizzydevil) [reply SConstruct:462]: Why do we need a key file? We're not building  a signed version. If we were, this would need to further above at line 400.

**Pull request merged**